pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/range/RangeIndexType.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pinot.segment.local.segment.index.range;
+
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.apache.pinot.segment.spi.ColumnMetadata;
+import org.apache.pinot.segment.spi.V1Constants;
+import org.apache.pinot.segment.spi.creator.IndexCreationContext;
+import org.apache.pinot.segment.spi.index.FieldIndexConfigs;
+import org.apache.pinot.segment.spi.index.IndexCreator;
+import org.apache.pinot.segment.spi.index.IndexHandler;
+import org.apache.pinot.segment.spi.index.IndexReader;
+import org.apache.pinot.segment.spi.index.IndexReaderFactory;
+import org.apache.pinot.segment.spi.index.IndexType;
+import org.apache.pinot.segment.spi.index.StandardIndexes;
+import org.apache.pinot.segment.spi.store.SegmentDirectory;
+import org.apache.pinot.spi.config.table.IndexConfig;
+import org.apache.pinot.spi.config.table.TableConfig;
+import org.apache.pinot.spi.data.Schema;
+
+
+public class RangeIndexType implements IndexType<IndexConfig, IndexReader, IndexCreator> {
+
+  /**
+   * The default range index version used when not specified in the TableConfig.
+   *
+   * This value should be equal to the one used in {@link org.apache.pinot.spi.config.table.IndexingConfig}
```

Comment:
is this comment for this INSTANCE constant? I feel it's not quite related, but I may have missed some details.

Reply:
Good catch and a funny story. I created this PR by using diff from changes in [`index-spi` branch](https://github.com/apache/pinot/pull/10184) and then removing things I didn't want to show in this PR. Here I removed a static constant but didn't removed the Javadoc.

The problem with my approach is that now if I delete the javadoc here, once I merge this with #10184 the comment will be removed without conflict and I may forget about it. Therefore what I'm going to do is to bring back the static constant. It is not going to be used in this PR, but it is going to ensure we don't accidentally remove the javadoc.